pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -755,27 +771,28 @@ func (pl *dynamicResources) PostFilter(ctx context.Context, cs *framework.CycleS
 		claim := state.claims[index]
 		if len(claim.Status.ReservedFor) == 0 ||
 			len(claim.Status.ReservedFor) == 1 && claim.Status.ReservedFor[0].UID == pod.UID {
-			// Before we tell a driver to deallocate a claim, we
-			// have to stop telling it to allocate. Otherwise,
-			// depending on timing, it will deallocate the claim,
-			// see a PodSchedulingContext with selected node, and
-			// allocate again for that same node.
-			if state.podSchedulingState.schedulingCtx != nil &&
-				state.podSchedulingState.schedulingCtx.Spec.SelectedNode != "" {
-				state.podSchedulingState.selectedNode = ptr.To("")
-				if err := state.podSchedulingState.publish(ctx, pod, pl.clientset); err != nil {
-					return nil, statusError(logger, err)
+			pl.fh.RunInBackground(ctx, pod, func(ctx context.Context) error {
+				logger := klog.FromContext(ctx)
+				// Before we tell a driver to deallocate a claim, we
+				// have to stop telling it to allocate. Otherwise,
+				// depending on timing, it will deallocate the claim,
+				// see a PodSchedulingContext with selected node, and
+				// allocate again for that same node.
+				if state.podSchedulingState.schedulingCtx != nil &&
+					state.podSchedulingState.schedulingCtx.Spec.SelectedNode != "" {
+					state.podSchedulingState.selectedNode = ptr.To("")
+					if err := state.podSchedulingState.publish(ctx, pod, pl.clientset); err != nil {
+						return err
+					}
 				}
-			}
 
-			claim := state.claims[index].DeepCopy()
-			claim.Status.DeallocationRequested = true
-			claim.Status.ReservedFor = nil
-			logger.V(5).Info("Requesting deallocation of ResourceClaim", "pod", klog.KObj(pod), "resourceclaim", klog.KObj(claim))
-			if err := state.updateClaimStatus(ctx, pl.clientset, index, claim); err != nil {
-				return nil, statusError(logger, err)
-			}
-			return nil, nil
+				claim := state.claims[index].DeepCopy()
+				claim.Status.DeallocationRequested = true
+				claim.Status.ReservedFor = nil
+				logger.V(5).Info("Requesting deallocation of ResourceClaim", "pod", klog.KObj(pod), "resourceclaim", klog.KObj(claim))
+				return state.updateClaimStatus(ctx, pl.clientset, index, claim)
+			})
+			return nil, framework.NewStatus(framework.Unschedulable, "waiting for deallocation of claim")
```

Comment:
I'm unsure why this returned a nil status before. I suspect it simply didn't matter for `PostFilter`. `Unschedulable` with a proper explanation seems better.